pkg/filesystem/directory.go

```diff
@@ -74,6 +74,8 @@ type Directory interface {
 
 	// Link is the equivalent of os.Link().
 	Link(oldName path.Component, newDirectory Directory, newName path.Component) error
+	// Clonefile is the equivalent of unix.CloneFile on macOS
```

Comment:
s/CloneFile/Clonefile/

Also make sure to end comments with a period, as the rest of the code base does this fairly consistently.